wgsl/index.bs

```diff
@@ -5709,6 +5712,377 @@ Note: If the function [=return value|returns a value=], that value is ignored.
+
+A <dfn export>behavior</dfn> is a set, whose elements may be:
+- Return
+- Discard
+- Break
+- Continue
+- Fallthrough
+- Next
+
+Each of those correspond to a way to exit a compound statement: either through a keyword, or by falling to the next statement ("Next").
+
+We note "*s*: *B*" to say that *s* respects the rules regarding behaviors, and has [=behavior=] *B*.
+
+For each function:
+- its body must be a valid statement by these rules
+- if it has a return type, its [=behavior=] must be one of {Return} or {Return, Discard}
+- otherwise, its [=behavior=] must be a subset of {Next, Return, Discard}
+
+We assign a [=behavior=] to each function: it is its body's [=behavior=] (treating the body as a regular statement), with any "Return" replaced by "Next".
+As a consequence of the rules above, a function behavior is always one of {Next}, {Discard} or {Next, Discard}.
+
+Similarly, we assign a [=behavior=] to each expression, since expressions can include function calls, which can discard.
+Like functions, expression behaviors are always one of {Next}, {Discard}, or {Next, Discard}.
+
+Note: There is currently no valid program with an expression that has a [=behavior=] of {Discard}.
+The reason is that only functions without a return type can have such a [=behavior=], and there is no compound expression in which such a function can be called.
+
+<table class='data'>
+  <caption>Rules for analyzing and validating the behaviors of statements</caption>
+  <thead>
+    <tr><th>Statement<th>Preconditions<th>Resulting behavior
+  </thead>
+  <tr>
+    <td class="nowrap">|s1| ; |s2|
+    <td class="nowrap">|s1|: |B1|<br>
+        Next in |B1|<br>
+        |s2|: |B2|
+    <td class="nowrap">|B1|&#x2216;{Next} &cup; |B2|
+  <tr>
+    <td class="nowrap">var x:T;
+    <td>
+    <td>{Next}
+  <tr>
+    <td class="nowrap">let x = |e|;
+    <td class="nowrap">|e|: |B|
+    <td>|B|
+  <tr>
+    <td class="nowrap">var x = |e|;
+    <td class="nowrap">|e|: |B|
+    <td>|B|
+  <tr>
+    <td class="nowrap">x = |e|;
+    <td class="nowrap">*e*: |B|
+    <td>|B|
+  <tr>
+    <td class="nowrap">|f|(|e1|, ..., |en|);
+    <td class="nowrap">|e1|: |B1|<br>
+        ...<br>
+        |en|: |Bn|<br>
+        |f|'s body has behavior |B|
+    <td class="nowrap">|B| &cup; (|B1| &cup; ... &cup; |Bn|)&#x2216;{Next}
+  <tr>
+    <td>return;
+    <td>
+    <td>{Return}
+  <tr>
+    <td class="nowrap">return |e|;
+    <td class="nowrap">|e|: |B|<br>
+    <td class="nowrap">|B|&#x2216;{Next} &cup; {Return}
+  <tr>
+    <td>discard;
+    <td>
+    <td>{Discard}
+  <tr>
+    <td>break;
+    <td>
+    <td>{Break}
+  <tr>
+    <td>continue;
+    <td>
+    <td>{Continue}
+  <tr>
+    <td>fallthrough;
+    <td>
+    <td>{Fallthrough}
+  <tr>
+    <td class="nowrap">if (|e|) |s1| elseif |s2| ...  else |sn|
```

Comment:
I had forgotten the fact that elseif has a condition. I reworked that part and added a note below, as well as an example.